### DISCUSSION TITLE
Accessing variables on Avalanche mainnet with web3 or ethers.js

### DISCUSSION THREAD

**Discussion by icegriffinguru** [web3together/web3together#19] [Q&A]
I've tried to access variables on avalanche network with web3 but failed.
There must be a problem with setting up blockchain network provider.
Here is the code.
```
const` Web3 = require('web3');
const url = 'https://speedy-nodes-nyc.moralis.io/4675ff3ce60e2f220ae956a7/avalanche/mainnet';
const provider = new Web3.providers.HttpProvider(url);
// console.log(provider);

const web3 = new Web3(provider);
// console.log(web3.eth);

web3.eth.getStorageAt('0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4', 0, console.log);
web3.eth.getStorageAt('0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4', 1, console.log);
web3.eth.getStorageAt('0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4', 2, console.log);
web3.eth.getStorageAt('0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4', 3, console.log);
```

I run the script three times and got three different results. I think there must be an error.
![image](https://user-images.githubusercontent.com/97042732/150680041-8f00be4c-0408-4cb9-913b-7d41473f5e9d.png)

**Comment by BunHouth** [comment]
@icegriffinguru Everything is look right. Result is difference because of display order.

Please check code below:

```js
import Web3 from 'web3';

const web3 = new Web3('https://speedy-nodes-nyc.moralis.io/xxxxxxxx/avalanche/mainnet');
web3.eth.getStorageAt("0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4", 0).then((position) => {
  console.log(`0 ${position}`)
});
web3.eth.getStorageAt("0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4", 1).then((position) => {
  console.log(`1 ${position}`)
});
web3.eth.getStorageAt("0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4", 2).then((position) => {
  console.log(`2 ${position}`)
});
web3.eth.getStorageAt("0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4", 3).then((position) => {
  console.log(`3 ${position}`)
});

```
<img width="784" alt="Screen Shot 2022-01-23 at 10 25 24 PM" src="https://user-images.githubusercontent.com/8031851/150680683-7a019ee6-65f8-4476-8f75-ac3b5be8cb22.png">

**Reply by icegriffinguru** [reply]
I've run the same code for three times but different results.
And your's is the same. Two results are different.
Do you think is it possible data on blockchain storage change every time?

**Reply by BunHouth** [reply]
Please check this code might make you more clear on response.

```js
import Web3 from 'web3';

const web3 = new Web3('https://speedy-nodes-nyc.moralis.io/xxxxxx/avalanche/mainnet');

const getStorageAt = async () => {
  await web3.eth.getStorageAt("0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4", 0, console.log)
  await web3.eth.getStorageAt("0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4", 1, console.log)
  await web3.eth.getStorageAt("0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4", 2, console.log)
  await web3.eth.getStorageAt("0x6c1c0319d8ddcb0ffe1a68c5b3829fd361587db4", 3, console.log)
}

getStorageAt();
```
<img width="877" alt="Screen Shot 2022-01-23 at 10 40 03 PM" src="https://user-images.githubusercontent.com/8031851/150681250-b95bea35-09a2-4311-8562-4eac5009d15b.png">

**Comment by icegriffinguru** [comment]
great, i got the same results.

![image](https://user-images.githubusercontent.com/97042732/150681703-cd63b7cb-c29e-40b7-a68a-3f43708f3e11.png)
